apps/meteor/server/models/raw/Analytics.ts

```diff
@@ -211,4 +211,117 @@ export class AnalyticsRaw extends BaseRaw<IAnalytic> implements IAnalyticsModel
 	findByTypeBeforeDate({ type, date }: { type: IAnalytic['type']; date: IAnalytic['date'] }): FindCursor<IAnalytic> {
 		return this.find({ type, date: { $lte: date } });
 	}
+
+	getRoomsWithNumberOfMessagesBetweenDateQuery({
+		types,
+		start,
+		end,
+		startOfLastWeek,
+		endOfLastWeek,
+		options,
+	}: {
+		types: Array<IRoom['t']>;
+		start: number;
+		end: number;
+		startOfLastWeek: number;
+		endOfLastWeek: number;
+		options?: any;
+	}) {
+		const typeAndDateMatch = {
+			$match: {
+				'type': 'messages',
+				'room.t': { $in: types },
+				'date': { $gte: startOfLastWeek, $lte: end },
+			},
```

Comment:
We already have an index on the `date` field for this collection :partying_face: We're using it on this query
Is your suggestion to add a compound index for these 3 fields? I'll try and check if we have any improvement

Reply:
Yeah, I'd suggest trying with some indexes and add the one that adds the best result with a `partialFilterExpresssion` on messages.

Maybe something like
```
room.t: 1, date: 1, partialFilterExpression: { type: 'messages'
```

and see if it gets better or not. 

Try with hints as well, as mongo would sometimes pick the index with the most data output so no docs are left out even if a better index is on the collection

Reply:
Done! Average time is much better now with the new index
We dont't need to have the `partialFilterExpression` here. `sparse` works just the same since only documents with `type: 'messages'` will have both fields.

Reply:
Nice!
About partial vs sparse: https://www.mongodb.com/docs/manual/core/index-partial/#comparison-with-sparse-indexes

Reply:
Ohhh, thanks. I'm now using a `partialFilterExpressions` instead
Their indexes are both the exact same size with my test data, but let's follow mongodb docs :pray: